src/helpers/amoc_xmpp_handlers.erl

```diff
@@ -57,12 +57,27 @@ stanza_handler(Pred, Action) ->
             end
     end.
 
+check_predicate(Pred, Stanza) when is_function(Pred, 1)->
+    apply_safely(Pred, [Stanza]).
+
 perform_action(Action, Client, Stanza, Metadata) when is_function(Action, 3) ->
-    Action(Client, Stanza, Metadata);
+    apply_safely(Action, [Client, Stanza, Metadata]);
 perform_action(Action, Client, Stanza, _Metadata) when is_function(Action, 2) ->
-    Action(Client, Stanza);
+    apply_safely(Action, [Client, Stanza]);
 perform_action(Action, _Client, _Stanza, _Metadata) when is_function(Action, 0) ->
-    Action().
+    apply_safely(Action, []).
+
+-spec apply_safely(function(), [term()]) -> {ok | error, term()}.
```

Comment:
`dialyzer`:
```erlang
Line 52 Column 17: The pattern 'true' can never match the type {'error',_} | {'ok',_}
Line 55 Column 17: The pattern 'false' can never match the type {'error',_} | {'ok',_}
```
Looking to logic, current function should return or `true` or `false`. See line `51`.

Reply:
yeah, I've seen it. will fix this one once.